io/ice/src/main/java/org/mobicents/media/io/ice/network/stun/StunHandler.java

```diff
@@ -111,8 +112,30 @@ private long extractPriority(StunRequest request) throws IllegalArgumentExceptio
 		 * The agent MUST use a short-term credential to authenticate the
 		 * request and perform a message integrity check.
 		 */
-		UsernameAttribute remoteUnameAttribute = (UsernameAttribute) request.getAttribute(StunAttribute.USERNAME);
-		String remoteUsername = new String(remoteUnameAttribute.getUsername());
+
+		// Produce Binding Response
+		TransportAddress transportAddress = new TransportAddress(remotePeer.getAddress(), remotePeer.getPort(), TransportProtocol.UDP);
+		StunResponse response = StunMessageFactory.createBindingResponse(request, transportAddress);
+		byte[] transactionID = request.getTransactionId();
+		try {
+			response.setTransactionID(transactionID);
+		} catch (StunException e) {
+			throw new IOException("Illegal STUN Transaction ID: " + new String(transactionID), e);
+		}
+
+		UsernameAttribute remoteUnameAttribute;
+		String remoteUsername;
+		// Send binding error response if username is null
+		try {
+			remoteUnameAttribute = (UsernameAttribute) request.getAttribute(StunAttribute.USERNAME);
+			remoteUsername = new String(remoteUnameAttribute.getUsername());
+		}
+		catch(NullPointerException nullPointer) {
```

Comment:
@farwaakhtar It's preferred to do a null-check if(remoteUnameAttribute == null) instead of catching an NPE like this.